API/models/productDao.js

```diff
@@ -0,0 +1,91 @@
+const appDataSource = require('./appDataSource');
+const { DatabaseError } = require('../utils/error');
+
+const getProductList = async (categoryId, sort, limit, offset) => {
+  try {
+    let whereCondition = '';
+
+    if (categoryId) {
+      whereCondition = `WHERE products.category_id IN (${categoryId})`;
+    }
+
+    const sortList = {
+      likedescending: 'likeCount DESC',
+      immediatebuyprice: 'immediateBuyPrice',
+      immediatesaleprice: 'immediateSalePrice DESC',
+      reviewdescending: 'reviewCount DESC',
+      premiumascending: 'premiumPercent',
+      premiumdescending: 'premiumPercent DESC',
+    };
+
+    const sortCondition = sortList[sort] || 'products.id';
```

Comment:
오름차순과 내림차순을 선택할 수 있는 인자도 같이 받으면 좋을 것 같습니다.

```suggestion
    const sortList = {
      likede: 'likeCount',
      immediatebuyprice: 'immediateBuyPrice',
      review: 'reviewCount',
      premium: 'premiumPercent',
    };
    const sortOrder = {
      desc: "DESC"
      asc: "ASC"
    }
    const sortCondition = sortList[sort] ? `${sortList[sort]} ${sortOrder[sortOrder]}` : 'products.id';
```